dotnet/src/Connectors/Connectors.AI.OpenAI/OpenAIKernelBuilderExtensions.cs

```diff
@@ -5,117 +5,114 @@
 using Microsoft.SemanticKernel.AI.Embeddings;
-using Microsoft.SemanticKernel.AI.ImageGeneration;
 using Microsoft.SemanticKernel.AI.TextCompletion;
 using Microsoft.SemanticKernel.Connectors.AI.OpenAI.ChatCompletion;
 using Microsoft.SemanticKernel.Connectors.AI.OpenAI.ImageGeneration;
 using Microsoft.SemanticKernel.Connectors.AI.OpenAI.TextCompletion;
 using Microsoft.SemanticKernel.Connectors.AI.OpenAI.TextEmbedding;
-using Microsoft.SemanticKernel.Reliability;
 
 #pragma warning disable IDE0130
 // ReSharper disable once CheckNamespace - Using NS of KernelConfig
 namespace Microsoft.SemanticKernel;
 #pragma warning restore IDE0130
 
-public static class KernelConfigOpenAIExtensions
+public static class OpenAIKernelBuilderExtensions
 {
     #region Text Completion
 
     /// <summary>
     /// Adds an Azure OpenAI text completion service to the list.
     /// See https://learn.microsoft.com/azure/cognitive-services/openai for service details.
     /// </summary>
-    /// <param name="config">The kernel config instance</param>
+    /// <param name="builder">The <see cref="KernelBuilder"/> instance</param>
     /// <param name="deploymentName">Azure OpenAI deployment name, see https://learn.microsoft.com/azure/cognitive-services/openai/how-to/create-resource</param>
     /// <param name="endpoint">Azure OpenAI deployment URL, see https://learn.microsoft.com/azure/cognitive-services/openai/quickstart</param>
     /// <param name="apiKey">Azure OpenAI API key, see https://learn.microsoft.com/azure/cognitive-services/openai/quickstart</param>
     /// <param name="serviceId">A local identifier for the given AI service</param>
+    /// <param name="setAsDefault">Whether the service should be the default for its type.</param>
     /// <param name="httpClient">Custom <see cref="HttpClient"/> for HTTP requests.</param>
-    /// <param name="logger">Application logger</param>
     /// <returns>Self instance</returns>
-    public static KernelConfig AddAzureTextCompletionService(this KernelConfig config,
+    public static KernelBuilder WithAzureTextCompletionService(this KernelBuilder builder,
         string deploymentName,
         string endpoint,
         string apiKey,
         string? serviceId = null,
-        HttpClient? httpClient = null,
-        ILogger? logger = null)
+        bool setAsDefault = false,
+        HttpClient? httpClient = null)
     {
-        ITextCompletion Factory(IKernel kernel) => new AzureTextCompletion(
-            deploymentName,
-            endpoint,
-            apiKey,
-            httpClient ?? kernel.Config.HttpHandlerFactory.CreateHttpClient(kernel.Log),
-            logger ?? kernel.Log);
-
-        config.AddTextCompletionService(Factory, serviceId);
+        builder.WithAIService<ITextCompletion>(serviceId, (logger) =>
+            new AzureTextCompletion(
+                deploymentName,
+                endpoint,
+                apiKey,
+                httpClient,
```

Comment:
`HttpHandlerFactory` -- what's the plan here?

Reply:
That's on @SergeyMenshykh's plate next. In any case, I expect these params will remain, as one must be able to override any default httpClient on a per-connector basis (i.e. maybe Azure OpenAI requires special access credentials, but OpenAI does not).

Reply:
Thanks Lee, I returned HttpHandlerFactory usage because we need default http client. The code might change a little bit or might not in a scope of the HTTP stack alignment user story and the parameters will remail as Shawn said.

Reply:
These changes will make `OpenAIHttpRetryPolicyTestAsync` fail (and therefore demo scenarios in the kernel syntax examples and elsewhere). I'm not positive why this test was marked as skip (since it was targeting OpenAI not Azure), but alas. I wonder if this should be staged somewhere before the follow up work is defined.

Reply:
#1096 in hopes to re-enable those tests.

Reply:
I rolled back the changes a few days ago, so the code doesn't have anything that should break the tests, at least I don't see. I'll reach you to get more details about it and discuss it. 
Regarding the retry for OpenAI, I plan to inject HttpClient instance with common retry handler to the OpeanAI connectors in scope of the Http stack improvements, so, if nothing changes during code review, the retry will be supported when the HTTP story is closed.